climakitae/selectors.py

```diff
@@ -56,62 +54,53 @@
-            "https://tigerweb.geo.census.gov/arcgis/rest/services/TIGERweb/State_County/MapServer/1/query?where=STATE='06'&f=geojson"
-        )
-        self._ca_counties = self._ca_counties.sort_values("NAME")
-
-        self._ca_watersheds_file = "https://gis.data.cnra.ca.gov/datasets/02ff4971b8084ca593309036fb72289c_0.zip?outSR=%7B%22latestWkid%22%3A3857%2C%22wkid%22%3A102100%7D"
-        self._ca_watersheds = gpd.read_file(self._ca_watersheds_file)
-        self._ca_watersheds = self._ca_watersheds.sort_values("Name")
+        self._cat = intake.open_catalog("https://cadcat.s3.amazonaws.com/parquet/catalog.yaml")
+        self._us_states = self._cat.states.read()
+        self._ca_counties = self._cat.counties.read().sort_values("NAME")
+        self._ca_watersheds = self._cat.huc8.read().sort_values("Name")
 
     def get_us_states(self):
         """
-        Opens regionmask to retrieve a dictionary of state abbreviations:index
+        Returns a custom sorted dictionary of state abbreviations and indices.
         """
-        _state_lookup = dict(
-            [
-                (
-                    abbrev,
-                    np.argwhere(np.asarray(self._us_states.abbrevs) == abbrev)[0][0],
-                )
-                for abbrev in [
-                    "CA",
-                    "NV",
-                    "OR",
-                    "WA",
-                    "UT",
-                    "MT",
-                    "ID",
-                    "AZ",
-                    "CO",
-                    "NM",
-                    "WY",
-                ]
-            ]
-        )
-        return _state_lookup
+        _states_subset_list = [
+            "CA",
+            "NV",
+            "OR",
+            "WA",
+            "UT",
+            "MT",
+            "ID",
+            "AZ",
+            "CO",
+            "NM",
+            "WY"
+        ]
+        _us_states_subset = self._us_states.query("abbrevs in @_states_subset_list")[["abbrevs"]]
+        _us_states_subset["abbrevs"] = pd.Categorical(_us_states_subset["abbrevs"], categories=_states_subset_list)
+        _us_states_subset.sort_values(by="abbrevs", inplace=True)
+
+        return dict(zip(_us_states_subset.abbrevs, _us_states_subset.index))
 
     def get_ca_counties(self):
         """
-        Returns a dictionary of California counties and their indices in the shapefile.
+        Returns a dictionary of California counties and their indices in the geoparquet file.
         """
         return pd.Series(
             self._ca_counties.index, index=self._ca_counties["NAME"]
         ).to_dict()
 
     def get_ca_watersheds(self):
         """
-        Returns a lookup dictionary for CA watersheds that references the shapefile.
+        Returns a lookup dictionary for CA watersheds that references the geoparquet file.
         """
         return pd.Series(
-            self._ca_watersheds["OBJECTID"].values, index=self._ca_watersheds["Name"]
```

Comment:
No longer using OBJECTID but using the internal index to conform with how the other subareas are retrieved.